### DISCUSSION TITLE
More than 2 extra pages

### DISCUSSION THREAD

**Discussion by Ajaksmaniac** [keycloakify/keycloakify#345] [Q&A] [answered]
Hi, As i understood, extra pages are like placeholders for the custom pages. And they could only be named extra page `my-extra-page-1.ftl` and my-extra-page-2.ftl. What should I do if I need more than 2 extra pages?

**Comment by garronej** [comment]
Hello @Ajaksmaniac,  

Lol of course they can be names whatever you want.  
Just edit the `package.json` https://docs.keycloakify.dev/build-options#keycloakify.extrapages  and the `src/keycloak-theme/login/kcContrext.ts`.  

Extra pages aren't placeholders for custom pages though.  If you need custom pages you can implement it however you want, your theme is a React App. No, extra pages are to be used when there is a specific Keycloak user facing pages that isn't supported by default that you want to customize. Todays there are verry few pages that aren't supported by default in the login theme but you could have a custom Keycloak plugin that adds one.   

Ref: https://docs.keycloakify.dev/limitations  

I hope it helps

**Reply by Ajaksmaniac** [reply]
Why i asked this question.

Created a custom keycloak plugin that at some point it opens a custom page called `login-sms.ftl`.(Worked before switching to keycloakify)

Started using keycloakify, rewrote my theme, created a page with pageId:'login-sms.ftl', registered it in package.json and src/keycloak-theme/login/kcContrext.ts, and when the page is supposed to be loaded i get this error "This app is a Keycloak theme It isn't meant to be deployed outside of Keycloak"

When i changed the pageId to `my-extra-page-1.ftl`, it worked like before.

**Reply by garronej** [reply]
Then yes `login-sms.ftl` is a good usecase for extra page.  
I think you aren't using the latest version of Keycloakify, there was a bug that was resolved not long ago about this.  
That or you didn't add `login-sms.ftl` in your package.json.  

If you're having issues I can have a look.

**Reply by garronej** [reply, marked as answer]
The issue I'm refering to: https://github.com/keycloakify/keycloakify/issues/334
